client/state/sharing/publicize/actions.js

```diff
@@ -145,10 +160,18 @@ export function fetchConnection( siteId, connectionId ) {
  * @returns {Function}                  Action thunk
  */
 export function createSiteConnection( siteId, keyringConnectionId, externalUserId ) {
-	return ( dispatch ) =>
-		wpcom
-			.undocumented()
-			.createConnection( keyringConnectionId, siteId, externalUserId, { shared: false } )
+	return ( dispatch ) => {
+		const body = {
+			keyring_connection_ID: keyringConnectionId,
+			shared: false,
+		};
+
+		if ( externalUserId ) {
+			body.external_user_ID = externalUserId;
+		}
+
+		return wpcom.req
+			.post( `/sites/${ siteId }/publicize-connections/new`, body )
```

Comment:
What about the case where `siteId` is null and the `/me/publicize-connections/new` should be used? Can it ever happen?

Reply:
I don't think it can happen, which is why I've cleaned it up. I've verified that all the sections that embed the primary connections component include the site selection middleware.